src/main/cpp/src/xxhash64.cu

```diff
@@ -271,9 +272,32 @@ hash_value_type __device__ inline XXHash_64<numeric::decimal128>::operator()(
 /**
  * @brief Computes the hash value of a row in the given table.
  *
+ * This functor uses Spark conventions for xxhash64 hashing, which differs from
+ * the xxhash64 implementation used in the rest of libcudf. These differences
+ * include:
+ * - Serially using the output hash as an input seed for the next item
+ * - Ignorance of null values
+ *
+ * The serial use of hashes as seeds means that data of different nested types
+ * can exhibit hash collisions. For example, a row of an integer column
+ * containing a 1 will have the same hash as a lists column of integers
+ * containing a list of [1] and a struct column of a single integer column
+ * containing a struct of {1}.
+ *
+ * As a consequence of ignoring null values, inputs like [1], [1, null], and
```

Comment:
[1] and [1, null] have the same hash value?
If item is null, then return the seed(e.g.: 42).

Reply:
> [1] and [1, null] have the same hash value? If item is null, then return the seed(e.g.: 42).

- For the first element, the seed is `42L`。
- For the other elements, the seed is the previous element's hash value.